resource/api/auth.py

```diff
@@ -35,65 +101,47 @@ def get_openid_configuration(issuer_url: str) -> dict:
     return response.json()
 
 
-def check_token(token: str, client_certificate: str) -> dict:
-    openid_config = get_openid_configuration(conf.ISSUER_URL)
-    introspection_endpoint = openid_config["introspection_endpoint"]
-    if (
-        "localhost" in introspection_endpoint
-    ):  # Bit of messing about for docker, consider bringing accounting app into the same project
-        introspection_endpoint = introspection_endpoint.replace(
-            "https://localhost:8000", conf.ISSUER_URL
-        )
-    log.debug("Token type", type(token))
-    try:
-        response = requests.post(
-            url=introspection_endpoint,
-            json={"token": token, "client_certificate": client_certificate},
-            auth=(conf.OAUTH_CLIENT_ID, conf.OAUTH_CLIENT_SECRET),
-            verify=False,
-        )
-        if response.status_code != 200:
-            log.error(f"introspection request failed: {response.text}")
-            raise AccessTokenValidatorError("Introspection request failed")
-
-    except requests.exceptions.RequestException as e:
-        log.error(f"introspection request failed: {e}")
-        raise AccessTokenValidatorError("Introspection request failed")
-    return response.json()
-
-
-def introspect(
-    client_certificate: str, token: str, x_fapi_interaction_id: Optional[str] = None
+def check_token(
+    client_certificate: str | None,
+    token: str,
+    x_fapi_interaction_id: Optional[str] = None,
```

Comment:
Follow the Fapi spec on headers - if an interaction is provided, use that, otherwise generate and add one to the response